storage/rocksdb/rdb_compact_filter.h

```diff
@@ -50,15 +50,12 @@ class Rdb_compact_filter : public rocksdb::CompactionFilter {
   void calculate_expiration_timestamp() const {
     assert(!m_expiration_timestamp.has_value());
 
-    uint64_t oldest_snapshot_timestamp = 0;
-    rocksdb::DB *const rdb = rdb_get_rocksdb_db();
-    if (!rdb->GetIntProperty(rocksdb::DB::Properties::kOldestSnapshotTime,
-                             &oldest_snapshot_timestamp) ||
-        oldest_snapshot_timestamp == 0) {
-      oldest_snapshot_timestamp = static_cast<uint64_t>(std::time(nullptr));
+    auto oldest_transaction_ts = oldest_transaction_timestamp();
+    if (oldest_transaction_ts == 0) {
+      oldest_transaction_ts = static_cast<uint64_t>(std::time(nullptr));
     }
 
-    m_expiration_timestamp = oldest_snapshot_timestamp;
+    m_expiration_timestamp = oldest_transaction_ts;
 
     if (rdb_is_binlog_ttl_enabled()) {
```

Comment:
I wonder if we can directly use the rocksdb_binlog_ttl_compaction_timestamp here instead of getting the min between two values.

Reply:
> I wonder if we can directly use the rocksdb_binlog_ttl_compaction_timestamp here instead of getting the min between two values.

We can't. Because if we have a long running transaction in secondary, the oldest_snapshot_timestamp can be older than the rocksdb_binlog_ttl_compaction_timestamp. We nee the oldest_snapshot_timestamp to prevent compacting away records needed in this long running transcation.